.github/workflows/test.yaml

```diff
@@ -0,0 +1,341 @@
+      # Note: Must run before get-engine-hash, which uses the virtualenv.
+      - name: Cache Pants Virtualenv
+        uses: actions/cache@v2
+        with:
+          path: |
+            build-support/virtualenvs
+          key: |
+            ${{ runner.os }}-pants-venv-${{ matrix.python-version }}-${{ hashFiles('pants/3rdparty/python/**', 'pants.toml') }}
+      - name: Get Engine Hash
+        id: get-engine-hash
+        run: |
+          echo "::set-output name=hash::$(./build-support/bin/rust/print_engine_hash.sh)"
+        shell: bash
+      - name: Cache Native Engine
+        uses: actions/cache@v2
+        with:
+          path: |
+            src/python/pants/engine/internals/native_engine.so
+            src/python/pants/engine/internals/native_engine.so.metadata
+          key: |
+            ${{ runner.os }}-engine-${{ steps.get-engine-hash.outputs.hash }}
+      - name: Set env vars
+        run: |
+          echo "PANTS_CONFIG_FILES=${{ github.workspace }}/pants.travis-ci.toml" >> ${GITHUB_ENV}
+      - name: Bootstrap Pants
+        run: |
+          ./pants --version
+      - name: Run smoke tests
+        run: |
+          ./pants help goals
+          ./pants list ::
+          ./pants roots
+          ./pants help targets
+      - name: Test and Lint Rust
+        # Note that running these in a separate job would require that job to rebuild
+        # the Rust code from scratch, since Cargo uses mtimes to invalidate, and those
+        # are not conserved across git clones. And since we have to build Rust code
+        # in this job anyway, and the tests only take a couple of minutes to run, we
+        # run them in the same job.
+        run: |
+          sudo apt-get install -y pkg-config fuse libfuse-dev
+          ./cargo clippy --all
+          # We pass --tests to skip doc tests because our generated protos contain invalid
+          # doc tests in their comments.
+          ./cargo test --all --tests -- --nocapture
+        if: "!contains(env.COMMIT_MESSAGE, '[ci skip-rust]')"
+      - name: Upload native_engine.so
+        uses: actions/upload-artifact@v2
+        with:
+          name: native_engine.so.${{ matrix.python-version }}.linux
+          path: |
+            src/python/pants/engine/internals/native_engine.so
+            src/python/pants/engine/internals/native_engine.so.metadata
+
+  test_python_linux:
+    name: Test Python (Linux)
+    runs-on: ubuntu-20.04
+    needs: bootstrap_pants_linux
+    strategy:
+      matrix:
+        python-version: [3.7.10]
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Cache Pants Virtualenv
+        uses: actions/cache@v2
+        with:
+          path: |
+            build-support/virtualenvs
+            ~/.cache/pants/plugins/
```

Comment:
I think this is now irrelevant with @stuhood's changes for plugin resolution - iiuc, that is now a standard Pants process cached in `lmdb_store` and `named_caches`.

Reply:
Good point, removed. We're not caching the `lmdb_store`, and I don't really want to since it grows without bound. This is what remote caching is for.